packages/mui-material/src/Badge/Badge.js

```diff
@@ -252,40 +236,60 @@ const Badge = React.forwardRef(function Badge(inProps, ref) {
     color = colorProp,
     overlap = overlapProp,
     anchorOrigin = anchorOriginProp,
+    variant = variantProp,
   } = invisible ? prevProps : props;
 
-  const ownerState = { ...props, anchorOrigin, invisible, color, overlap };
+  const ownerState = { ...props, anchorOrigin, invisible, color, overlap, variant };
   const classes = extendUtilityClasses(ownerState);
 
+  let displayValue;
+
+  if (variant !== 'dot') {
+    displayValue =
```

Comment:
This is duplicated with the logic inside `useBadge`. Should we maybe just propagate null/undefined in the `badgeContent` if the variant is dot? I don't think there is need to process the `displayValue` here.

Reply:
I removed this logic from `useBadge` - it doesn't know about the variant anymore as it's a purely visual prop.